core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -187,6 +192,60 @@ void SYCLInternal::finalize() {
   m_queue.reset();
 }
 
+void* SYCLInternal::scratch_space(
+    const Kokkos::Experimental::SYCL::size_type size) {
+  const size_type sizeScratchGrain =
+      sizeof(Kokkos::Experimental::SYCL::size_type);
+  if (verify_is_initialized("scratch_space") &&
+      m_scratchSpaceCount * sizeScratchGrain < size) {
+    m_scratchSpaceCount = (size + sizeScratchGrain - 1) / sizeScratchGrain;
+
+    using Record = Kokkos::Impl::SharedAllocationRecord<
+        Kokkos::Experimental::SYCLDeviceUSMSpace, void>;
```

Comment:
Wait did you mean to allocate in device memory?
```suggestion
    using Record = Kokkos::Impl::SharedAllocationRecord<
        Kokkos::Experimental::SYCLSharedUSMSpace, void>;
```

Reply:
Yes. Do you see any problem with that?

Reply:
That you don't clearly state it in the description.

Reply:
From a slack discussion (after this comment):
> IIRC, we needed that [shared memory] in an early implementation where we were sometimes assigning directly instead of going through `deep_copy` to get the result back to `result_ptr`.